core/src/main/scala/org/apache/spark/scheduler/cluster/CoarseGrainedSchedulerBackend.scala

```diff
@@ -404,7 +405,7 @@ class CoarseGrainedSchedulerBackend(scheduler: TaskSchedulerImpl, val rpcEnv: Rp
         val ugi = UserGroupInformation.getCurrentUser()
         val tokens = if (dtm.renewalEnabled) {
           dtm.start()
-        } else if (ugi.hasKerberosCredentials()) {
+        } else if (ugi.hasKerberosCredentials() || ugi.getAuthenticationMethod == PROXY) {
```

Comment:
Use `SparkHadoopUtil.get.isProxyUser`.

I think I need to test this, though. `dtm.obtainDelegationTokens(creds)` will probably fail if you are running as proxy user but does not have access to the real UGI - e.g. submitting an app in cluster mode using a proxy user.

Reply:
Tested successfully in my environment. Both client mode and cluster mode using a proxy user.